src/Routes/seasons.ts

```diff
@@ -0,0 +1,31 @@
+﻿/*
+ * Seasons controller
+ */
+//import * as assert from "assert";
+import * as express from "express";
+import { SqlDataProvider } from "../DataProviders/SqlDataProvider/SqlDataProvider";
+import { Configuration } from "../Common/Configuration";
+import { Networking } from "../Common/Helpers";
+import { IDataProvider } from "../DataProviders/IDataProvider";
+const router = express.Router();
+
+// TODO: Add a factory pattern
```

Comment:
better than factory - can you use dependency injection? as your application grows, a factory will become hard to manage (and especially to test)

Reply:
Opened issue #9 to address this.